src/app/public/modules/angular-tree/angular-tree-node.component.ts

```diff
@@ -0,0 +1,221 @@
+    if (!this.skyAngularTreeWrapper) {
+      console.warn(`<sky-angular-tree-node-wrapper> should be wrapped inside a <sky-angular-tree-wrapper> component.`);
+    }
+
+    // Because we're binding the checkbox to node's children properties, we need to manually control change detection.
+    // Here, we listen to the tree's state and force change detection in the setters if the value has changed.
+    this.node.treeModel.subscribeToState((state: ITreeState) => {
+      this.isSelected = this.node.isSelected;
+      this.isPartiallySelected = this.node.isPartiallySelected;
+
+      if (state.focusedNodeId) {
+        this.focused = state.focusedNodeId === this.node.id;
+      }
+    });
+
+    // Make the first root node tabbable.
+    if (this.node.isRoot && this.node.index === 0) {
+      this.tabIndex = 0;
+    }
+  }
+
+  public ngAfterViewInit(): void {
+    // Wait for node to render, then reset all child tabIndexes to -1.
+    setTimeout(() => {
+      this.focusableChildren = this.adapterService.getFocusableChildren(this.nodeContentWrapperRef.nativeElement);
+      this.adapterService.setTabIndexOfFocusableElems(this.nodeContentWrapperRef.nativeElement, -1);
+    }, 1000);
+  }
+
+  // If single-select, set aria-selected=true for the selected node and undefined for all the others.
+  // For multiple-select, set aria-selected to either true or false.
+  // If node cannot be selected, aria-selected should be undefined (e.g. parent nodes in leaf-select-only mode).
+  public ariaSelected(): boolean {
+    if (!this.skyAngularTreeWrapper) {
+      return;
+    }
+
+    if (this.skyAngularTreeWrapper.selectSingle) {
+      return this.isSelected ? true : undefined;
+    }
+
+    if (!this.isSelectable()) {
+      return;
+    }
+
+    return !!this.isSelected;
+  }
+
+  public showCheckbox(): boolean {
+    // Check for checkbox mode enabled, but also respect leaf-node and single-select settings.
+    return this.node.options.useCheckbox && this.isSelectable() && !this.skyAngularTreeWrapper.selectSingle;
+  }
+
+  public showSelectedClass(): boolean {
+    return this.isSelectable() && this.isSelected && !this.isPartiallySelected;
+  }
+
+  public showActiveClass(): boolean {
+    return this.node.isActive && !this.node.treeModel.options.useCheckbox;
+  }
+
+  public onFocus(): void {
+    this.node.treeModel.setFocus(true);
+    this.node.focus();
+    this.childFocusIndex = undefined;
+  }
+
+  // Cycle backwards through interactive child elements
+  // If user reaches the beginning, activate drill up.
+  public onArrowLeft(event: KeyboardEvent): void {
+    /* istanbul ignore else */
+    if (document.activeElement === event.target) {
+      if (this.childFocusIndex !== undefined) {
+        if (this.childFocusIndex === 0) {
+          this.childFocusIndex = undefined;
+        } else {
+          this.childFocusIndex--;
+        }
+      } else {
+        this.node.treeModel.focusDrillUp();
+      }
+      event.stopPropagation();
```

Comment:
Would this suffer from scrolling if the tree view is on a page where there is more content than what will fit left and right? Should we prevent default like we did on the reorder stuff?

Reply:
Yeah, I think we should to be safe. Good thought.